indico/modules/events/schemas.py

```diff
@@ -5,14 +5,56 @@
 # modify it under the terms of the MIT License; see the
 # LICENSE file for more details.
 
-from marshmallow import fields
+import re
+
+from marshmallow import ValidationError, fields, validates_schema
+from marshmallow.validate import Length
 
 from indico.core.marshmallow import mm
 from indico.modules.events.models.events import Event
 from indico.modules.events.models.principals import EventPrincipal
+from indico.util.i18n import _
 from indico.util.marshmallow import PrincipalPermissionList
 
 
+class ValidRegExField(fields.Field):
+    """A valid Python regular expression."""
+
+    def _serialize(self, value, attr, obj, **kwargs):
+        return value.pattern
+
+    def _deserialize(self, value, attr, data, **kwargs):
+        if len(value) < 2:
+            raise ValidationError(_('Empty regex'))
+        try:
+            return re.compile(value)
```

Comment:
i'd return the string instead of the compiled regex (see my other comment elsewhere in this PR)